File: 75.js
function longestConsecutive(nums) {
    const setNums = new Set(nums);
    let longest = 0;

    for (const n of nums) {
        if (!setNums.has(n - 1)) {
            let cur = n;
            let len = 1;
            while (setNums.has(cur + 1)) {
                cur++;
                len++;
            }
            longest = Math.max(longest, len);
        }
    }
    return longest;
}

File: 75.py
def longest_consecutive(nums):
    num_set = set(nums)
    longest = 0

    for n in nums:
        if n - 1 not in num_set:  # start of sequence
            length = 1
            cur = n
            while cur + 1 in num_set:
                cur += 1
                length += 1
            longest = max(longest, length)
    
    return longest
